pkg/dispatcher/workerpools/controller.go

```diff
@@ -106,6 +106,10 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 
 				bulk := &bulkv1.CalculationBulk{}
 				if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: firstBulk.Name}, bulk); err != nil {
+					delete(workerpool.Spec.CalculationBulks, firstBulk.Name)
+					if errUpdate := r.client.Update(ctx, workerpool); errUpdate != nil {
```

Comment:
Well, some points:

- We need to do this operation only when the bulk is deleted. Here we delete the calc on all errors.
- In order to update the worker pool we need to retry when there is a conflict.
- You are trying to resolve the problem in the wrong controller. The bulk controller should take care of this.

Reply:
In the bulk controller, we only get object, that actually exist - we want them to stay. How can I then remove the unwanted bulks from workerpools. 
Lets say we have 0 active calculation bulks. We have an old calculation bulk stuck in workerpools - bulk-A. Now we create a bulk-B and start a reconcile function in bulks controller. There is no way i can delete the stuck bulk-A. The reconcile function from bulk controller gets bulk-B - it only gets the active bulk objects obviously. Am I really supposed to be solving
this in bulk controller? Also same thing is happening for calculations. @droslean